### DISCUSSION TITLE
NEXTAUTH_URL don't work

### DISCUSSION THREAD

**Discussion by pogran** [nextauthjs/next-auth#3635] [Help]
### Question 💬

Why do you have wrong desc in your doc? https://next-auth.js.org/configuration/options#nextauth_url , this params NEXTAUTH_URL dont work. In my app I have different basePath

```
next.config.js
module.exports = {
  basePath: '/admin',
};
```
I don't understand how to fix problem with path. I got error /api/auth/session don't found. And after one hour I only found how to fix it. need to add in SessionProvider basePath="/admin/api/auth" . Why in your doc I didn't find info about this moment?

### How to reproduce ☕️

```
<SessionProvider
     session={pageProps.session} >
      basePath="/admin/api/auth"
>
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by boutterudy** [comment]
After 2 days of research, I finally came across your question. Thank you @pogran for posting your solution, because without you I would still be stuck with the same problem.

The documentation should be updated, it's a shame because NextAuth is really good, the fact that the documentation is not complete greatly complicates the use of this powerful authentication solution.

I completely agree with you on the idea of adding this information to the official documentation.

**Reply by balazsorban44** [reply]
If you find an error/something missing in the docs, you can open a PR to https://github.com/nextauthjs/docs

There is also an edit button on the bottom of each page, so you don't even have to leave the browser.

![image](https://user-images.githubusercontent.com/18369201/149684744-23035bf1-98a7-45d3-92dd-4a4fcc9f191e.png)

**Reply by boutterudy** [reply]
As you've suggested, I've openned a PR to add more informations about usage of custom base path : https://github.com/nextauthjs/docs/pull/203

**Reply by roshan-ican** [reply]
I still have not been able to solve the problem yet, like I am not understanding it maybe my problem is similar to what you guys are facing, whenever I log in with my credentials it should redirect me to the Home page and the name should be been in the login button.
Please help

**Reply by boutterudy** [reply]
Can you please tell us more about your issue?

You're using a basePath ? And login doesn't work ?

If yes, I suggest you to follow [these instructions](https://stackoverflow.com/questions/70717224/how-to-define-a-custom-base-path-when-nextauth-url-doesnt-work#answer-70733866) and check the documentation [here](https://next-auth.js.org/getting-started/client#custom-base-path) and [here](https://next-auth.js.org/configuration/options#nextauth_url).

**Reply by NikitaKlimuk** [reply]
I don't quite understand how to solve this problem, I've tried all possible options, including those described above, but it still can't solve the redirect problem.

I set it to NEXTAUTH_URL=sebsite/next/api/auth
then I put it in SessionProvider basePath="/next/api/auth".

and I get errors because next auth can't find anything at this address. What am I doing wrong and how should it work?
GET website/next/api/auth/session 404 (Not Found)
client.js:1 [next-auth][error][CLIENT_FETCH_ERROR]
